eo-parser/src/main/java/org/eolang/parser/XeListener.java

```diff
@@ -347,8 +360,10 @@ public void enterApplicable(final ProgramParser.ApplicableContext ctx) {
             if (ctx.STAR() != null) {
                 base = "tuple";
                 this.objects.prop("data", "tuple");
+                this.objects.oprop(this.versioned, "ver");
```

Comment:
@maxonfjvipon If we add `ver` attribute for all objects without checking `versioned`, will it break something?

Reply:
@volodya-lombrozo I guess so. For example in `DiscoverMojo` we have the next code:
```java
private Collection<String> names(final XML xml, final String tojo) {
        return new SetOf<>(
            new Mapped<>(
                (String name) -> this.versioned(name, tojo).toString(),
                new Filtered<>(
                    name -> !name.isEmpty(),
                    xml.xpath(
                        String.join(
                            "",
                            "//o[",
                            "not(starts-with(@base,'.'))",
                            " and @base != 'Q'",
                            " and @base != '^'",
                            " and @base != '$'",
                            " and @base != '&'",
                            " and not(@ref)",
                            "]/string-join((@base,@ver),'",
                            OnReplaced.DELIMITER,
                            "')"
                        )
                    )
                )
            )
        );
    }
```
It does not check `withVersions` flag and joins `base` and `ver`. If there will be many empty `ver` attributes, there will be troubles.

Reply:
@volodya-lombrozo so you suggest replace one set of changes, which:
- is safer
- is more controllable (because it's just a boolean variable)
- lets us do not have extra attributes where they are not needed
with other set of changes, that works for now, but uncontrollable and will lead to the cases, where we will have eo code without any hint to the versions, but there will be some confusing emtpy "ver" attributes in xmir.

Reply:
@volodya-lombrozo if you personally don't like `oprop` method, I can remove it and replace its occurrences with its content in `XeListener`

Reply:
@maxonfjvipon 
1. Which type of 'safety' are you referring to? I run the tests and showed that everything works [fine](https://github.com/objectionary/eo/pull/2501/files#r1332960989).
2. Why do we need the control here? Moreover, as I already mentioned, the solution with boolean values creates even more problems (like [that](https://www.yegor256.com/2016/04/19/object-must-not-be-configurable.html) and [OCP](https://en.wikipedia.org/wiki/Open–closed_principle) violation).
3. The current implementation of `XeListener` in that PR does exactly what you mentioned: it adds empty "ver" attributes.